### ISSUE TITLE
[Feature Proposal] Support CSS Logical Properties

### ISSUE THREAD

**Issue by simonhermann** [tailwindcss/tailwindcss#949]
Directions like `right`, `left`, `bottom`, `top`, used in properties like `margin`, `padding` or `border` only make sense in left-to-right + top-to-bottom (`horizontal-tb`) writing mode.

Now we have new way of writing these, independently of writing mode.

Quote from [caniuse.com](https://caniuse.com/#search=logical%20properties):

>Logical properties and values provide control of layout through logical, rather than physical, direction and dimension mappings. These properties are writing-mode relative equivalents of their corresponding physical properties.

The logical directions become more and more common, for example are they used with flexbox or grid, so applying them to all CSS properties totally makes sense and will surely become the new standard.

Details how it works can be found in the [W3 Spec](https://www.w3.org/TR/css-logical-1/).

TLDR: 
vertical direction = inline flow
horizontal direction = block flow
start = where you start writing
end = where you end writing

so in default writing mode `horizontal-tb`:

`text-align: left;` becomes `text-align: start;`
`float: right;` becomes `text-align: end;`
`margin-right` becomes `margin-inline-end`
`padding-top` becomes `padding-inline-start`
`border-bottom` becomes `border-block-start`
and so on.

Following the tailwind naming scheme, the functional tailwind classes could be:

`m{i|b}{s|e}-{size}`  --> e.g. `mis-1` (better known as `ml-1`)

`text-{i|b}{s|e}` or `text-{inline|block}-{start|end}` 🤔 

Since this is not as intuitive to many users I and and the physical directions are not going to be depreciated soon, guess these just be an additional set of classes instead of replacing the current ones.

What do you think?

**Comment by hacknug** [comment]
Please! Agree this is is something core should do something about. I vaguely mentioned it in [#807](https://github.com/tailwindcss/tailwindcss/pull/807#issuecomment-477586459) and [#876](https://github.com/tailwindcss/tailwindcss/pull/876#issuecomment-487111534).

Interested on reading everyone else's thoughts on this (especially, people building things for non ltr-ttb languages).

**Comment by tlgreg** [comment]
I had almost the same naming convention with a plugin experiment I had. The only issue with it is the vertical margin/padding which would be the same as the bottom margin/padding. `mb-{size}`
I eventually went with two letters for the short form of the axes (`bl` and `il`).

There is also a conflict with insets, as from v1 there is an actual inset class, there is no difference in naming with logical/physical dimensions.

**Comment by Moeriki** [comment]
Two more options to avoid conflicting margin-bottom and margin-block acronyms.

Revert the acronym.

- `im`: Inline Margin
- `ism`: Inline Start Margin
- `iem`: Inline End Margin
- `bm`: Block Margin
- `bsm`: Block Start Margin
- `bem`: Block End Margin

Prefix with `l` for logical.

- `lmi`: Logical Margin Inline
- `lmis`: Logical Margin Inline Start
- `lmie`: Logical Margin Inline End
- `lmb`: Logical Margin Block
- `lmbs`: Logical Margin Block Start
- `lmbe`: Logical Margin Block End

**Comment by adamwathan** [comment]
I'd rather this be developed as a plugin first and if it gets enough traction we can explore adding it to core. Hopefully by then the Chromium version of Edge will have replaced the current version so the browser support will be better as well.

**Comment by omarkhatibco** [comment]
Hi eveyone, 

as Adam (@adamwathan) recommended I made this proposal as a plugin, your feedback and review is appreciated!

[Github](https://github.com/omarkhatibco/tailwind-css-logical-properties) - [NPM](https://www.npmjs.com/package/tailwind-css-logical-properties)

**Comment by Youhan** [comment]
Are there any updates on this? I just think `text-start` and `text-end` make more sense than `text-left` and `text-right`. I assume the benefits are obvious!

**Comment by Youhan** [comment]
Oh! For anyone landing here, looks like at the moment this plugin is the way to go: https://github.com/stevecochrane/tailwindcss-logical

**Comment by yaronuliel** [comment]
It has been more than 2.5 years since this issue was closed, and the support for CSS logical properties in all [supported browsers](https://tailwindcss.com/docs/browser-support) is pretty much complete now. 
Considering that, and the fact that on v3 we have amazing purge support, out of the box - I think it is time to reconsider adding CSS logical properties as core utilities rather than an external community maintained plugin.
